agent/service-message-maven-extension/src/main/java/nu/studer/teamcity/buildscan/agent/servicemessage/BuildScanServiceMessageDevelocityListener.java

```diff
@@ -0,0 +1,35 @@
+package nu.studer.teamcity.buildscan.agent.servicemessage;
+
+import com.gradle.develocity.agent.maven.api.DevelocityApi;
+import com.gradle.develocity.agent.maven.api.DevelocityListener;
+import org.apache.maven.execution.MavenSession;
+import org.codehaus.plexus.component.annotations.Component;
+
+import java.net.URI;
+import java.util.function.BiConsumer;
+
+@SuppressWarnings("unused")
+@Component(
+    role = DevelocityListener.class,
+    hint = "build-scan-service-message-develocity",
+    description = "Interacts with Maven builds invoked from TeamCity"
+)
+public final class BuildScanServiceMessageDevelocityListener extends BuildScanServiceMessageListener implements DevelocityListener {
+
+    @Override
+    public void configure(DevelocityApi api, MavenSession session) {
+        super.configure(new BuildScanApiAdapter() {
+            @Override
+            public void publishOnDemand() {
+                // publication is controlled via a system property in this case
+                api.getBuildScan().publishing(p -> p.onlyIf(ctx -> false));
```

Comment:
Is this the same as publish on demand? Or does `p.onlyIf(ctx -> false)` mean that a build scan would never be published, even if `-Dscan` is passed?

Reply:
This is now removed in favor of the public adapters library: https://github.com/etiennestuder/teamcity-build-scan-plugin/pull/373/commits/08f2e9924c13e98aa86775b63179ea0d298289f9

See https://github.com/gradle/develocity-agent-adapters